src/Private/Helpers.ps1

```diff
@@ -1714,37 +1736,58 @@ function Test-PodePath {
         $NoStatus,
 
         [switch]
-        $FailOnDirectory
+        $FailOnDirectory,
+
+        [switch]
+        $Force,
+
+        [switch]
+        $ReturnItem
     )
 
-    # if the file doesnt exist then fail on 404
-    if ([string]::IsNullOrWhiteSpace($Path) -or !(Test-Path $Path)) {
-        if (!$NoStatus) {
-            Set-PodeResponseStatus -Code 404
-        }
+    $statusCode = 404
 
-        return $false
-    }
+    if (![string]::IsNullOrWhiteSpace($Path)) {
+        try {
+            $item = Get-Item $Path -Force:$Force -ErrorAction Stop
+            if (($null -ne $item) -and (!$FailOnDirectory -or !$item.PSIsContainer)) {
+                $statusCode = 200
+            }
+        }
+        catch [System.Management.Automation.ItemNotFoundException] {
+            $statusCode = 404
+        }
+        catch [System.UnauthorizedAccessException] {
+            $statusCode = 401
+        }
+        catch {
+            $statusCode = 400
+        }
 
-    # if the file isn't accessible then fail 401
-    if (!(Test-PodePathAccess $Path)) {
-        if (!$NoStatus) {
-            Set-PodeResponseStatus -Code 401
+        if ($statusCode -eq 200) {
+            if ($ReturnItem) {
+                return $item
+            }
+            return $true
         }
+    }
 
-        return $false
+    # if we failed to get the file, report back the status code and/or return true/false
+    if (!$NoStatus.IsPresent) {
+        Set-PodeResponseStatus -Code $statusCode
     }
 
-    # if we're failing on a directory then fail on 404
-    if ($FailOnDirectory -and (Test-PodePathIsDirectory $Path)) {
-        if (!$NoStatus) {
-            Set-PodeResponseStatus -Code 404
+    if ($ReturnItem) {
+        if ($statusCode -eq 200) {
+            return $item
+        }
+        else {
+            return $null
         }
-
-        return $false
     }
 
-    return $true
+    return ($statusCode -eq 200)
```

Comment:
The 200 status code parts at the bottom won't be hit, as the `try` returns beforehand. A 200 doesn't really need to be explicitly set, as Pode does this by default as it's default response status unless a different one is set by `Set-PodeResponseStatus`.

This could then be condensed to the following, moving the try success return further up:
```powershell
$statusCode = 404

if (![string]::IsNullOrWhiteSpace($Path)) {
    try {
        $item = Get-Item $Path -Force:$Force -ErrorAction Stop
        if (($null -ne $item) -and (!$FailOnDirectory -or !$item.PSIsContainer)) {
            if ($ReturnItem) {
                return $item
            }
            return $true
        }
    }
    catch [System.Management.Automation.ItemNotFoundException] {
        $statusCode = 404
    }
    catch [System.UnauthorizedAccessException] {
        $statusCode = 401
    }
    catch {
        $statusCode = 400
    }
}

# if we failed to get the file, report back the status code and/or return true/false
if (!$NoStatus.IsPresent) {
    Set-PodeResponseStatus -Code $statusCode
}

if ($ReturnItem) {
    return $null
}
return $false
```

Reply:
because I forgot to delete it :D

Reply:
The issue the last commit fixed, due to calling `Set-PodeResponseStatus` in `Test-PodePath`, is because you're setting a 200 response within the function, when it's not needed - the original version of this function doesn't set a 200, a it can be left to Pode's default. If you use the code snippet above everything works as expected, even the tests.

Reply:
I understood the issue thanks. I'll fix morning

Reply:
done